test/distributed/org/apache/cassandra/distributed/impl/MessageFilters.java

```diff
@@ -87,70 +86,80 @@ public boolean equals(Object that)
         public boolean equals(Filter that)
         {
             return Arrays.equals(from, that.from)
-                    && Arrays.equals(to, that.to)
-                    && Arrays.equals(verbs, that.verbs);
-        }
-
-        public boolean matches(int from, int to, int verb)
-        {
-            return (this.from == null || Arrays.binarySearch(this.from, from) >= 0)
-                    && (this.to == null || Arrays.binarySearch(this.to, to) >= 0)
-                    && (this.verbs == null || Arrays.binarySearch(this.verbs, verb) >= 0);
+                   && Arrays.equals(to, that.to)
+                   && Arrays.equals(verbs, that.verbs);
         }
 
-        public Filter restore()
+        public Filter off()
         {
             filters.remove(this);
             return this;
         }
 
-        public Filter drop()
+        public Filter on()
         {
             filters.add(this);
             return this;
         }
+
+        public boolean matches(int from, int to, IMessage msg)
+        {
+            return (this.from == null || Arrays.binarySearch(this.from, from) >= 0)
+                   && (this.to == null || Arrays.binarySearch(this.to, to) >= 0)
+                   && (this.verbs == null || Arrays.binarySearch(this.verbs, msg.verb()) >= 0)
+                   && (this.matcher == null || this.matcher.matches(from, to, msg));
+        }
     }
 
     public class Builder implements IMessageFilters.Builder
     {
         int[] from;
         int[] to;
         int[] verbs;
+        Matcher onInstance;
```

Comment:
nit, since we no longer transfer this is no longer `onInstance`

Reply:
+1